src/components/pages/roster-page/edit-roster-entry-dialog.tsx

```diff
@@ -119,22 +119,30 @@ export const EditRosterEntryDialog = (props: EditRosterEntryDialogProps) => {
 
     const requiredColumns = rosterColumnInfos.filter(columnInfo => columnInfo.required);
     if (requiredColumns) {
+      const regex = RegExp(/^[0-9]{10}$/g);
+
       for (const column of requiredColumns) {
+        const type = column.type;
+        const name = column.name;
+        const value = rosterEntry[name] as string;
+
+        if (value == null) {
+          result = false;
+          break;
+        }
 
-        if (rosterEntry[column.name] == null) {
-          return false;
+        if ((type === ApiRosterColumnType.String
+          || type === ApiRosterColumnType.Date
+          || type === ApiRosterColumnType.DateTime)
+          && (value.length === 0 || value.trim().length === 0)
+        ) {
+          result = false;
+          break;
         }
 
```

Comment:
I changed the `switch` statement to `if` so we can `break` out of the loop on an invalid value. Otherwise, the loop continues through the remaining values.

Reply:
I don't think we actually even need that `result` variable. It doesn't look like `result` can ever get set to true again once it's set to false, so we can potentially just do `return false` instead of doing `result = false; break;`, and then put `return true` at the end of the function. Since there's no need to break here we could also get away with a `switch`, but the `if` works fine too.

Reply:
Removed `result` variable and reverted to `switch` in [latest commit](https://github.com/mystatus/covid19-reports/pull/56/commits/2600dfe7db5443f47c8c15a3a3c69fb8c6611e70).